test/MLJ/unit.jl

```diff
@@ -186,6 +195,104 @@ end
+
+        suggestions = Dict{Symbol}[]
+        tuning, state = setup(;n_startup=3,space=single_space, suggest=suggestions)
+
+        output, _ = MLJTuning.models(tuning, testGenericKwargModel, nothing, state, 0, 0)
+
+        @test output isa Vector{<:Tuple{Any, TreeParzen.Trials.Trial}}
+        @test length(output) == 3
+        param_x_values = getindex.(getproperty.(last.(output), :hyperparams), :x)
+        # given a defined space with uniform distribution between 6. and 10.
+        # the tests below checks that the param x from the dummy model was updated
+        # in each trial within that range
+        @test minimum(param_x_values) > 6.
+        @test maximum(param_x_values) < 10.
+        # checking that default was updated although it is possible to have an odd case where
+        # all results are 5.
+        @test !all(in.(param_x_values, [(5., 5., 5.)]))
+
+    end
+
+    @testset "no suggestions with a generic kwarg model constructor example --list of Dicts" begin
+
+        suggestions = Dict{Symbol}[]
+        tuning, state = setup(;n_startup=3,space=joint_space, suggest=suggestions)
+
+        output, _ = MLJTuning.models(tuning, testGenericKwargModel, nothing, state, 0, 0)
+
+        @test output isa Vector{<:Tuple{Any, TreeParzen.Trials.Trial}}
+        @test length(output) == 3
+        joint_values = getindex.(getproperty.(last.(output), :hyperparams), :joint)
+        # checks that parameters x and y were updated within their uniform distributions
+        # and that they differ from their defaults
+        for d in joint_values
+            if haskey(d, :x)
+                @test all(6. .<= d[:x] .<= 10.)
+                @test !all(in.(d[:x], [(5., 5., 5.)]))
+            end
+            if haskey(d, :y)
+                @test all(-3. .<= d[:y] .<= 3.)
+                @test !all(in.(d[:y], [(1., 1., 1.)]))
+            end
+        end
+
+    end
+
+    @testset "no suggestions with a generic kwarg model constructor example --conditioned choices" begin
+
+        suggestions = Dict{Symbol}[]
+        tuning, state = setup(;n_startup=3,space=nested_space, suggest=suggestions)
+
+        output, _ = MLJTuning.models(tuning, testGenericKwargModel, nothing, state, 0, 0)
+
+        @test output isa Vector{<:Tuple{Any, TreeParzen.Trials.Trial}}
+        @test length(output) == 3
+        param_z_values = getindex.(getproperty.(last.(output), :hyperparams), :z)
+        # checks that all parameters were updated within their uniform distributions
+        # and that they differ from their defaults
+        for d in param_z_values
+            if haskey(d, :x)
+                @test all(6. .<= d[:x] .<= 10.)
+                @test !all(in.(d[:x], [(5., 5., 5.)]))
+            end
+            if haskey(d, :y)
+                @test all(-3. .<= d[:y] .<= 3.)
+                @test !all(in.(d[:y], [(1., 1., 1.)]))
+            end
+            if haskey(d, :z)
+                @test any(in.(d[:z], [(true, false)]))
```

Comment:
I wanted this to show that param `:z` differs from its default `false` value but given the small number of draws, it is highly possible for the 3 outputs to return false so `all(in.(d[:z], [(false, false, false)]))` might occasionally fail. So this tests just checks z is returned and contains any of the options.

Reply:
makes sense this way we test all 3 `:z` parameter changes depending on whether values for `:x`, `:y` or none were chosen